### PR TITLE
atc(fix): fix an authorization bug, when a user has multiple roles in a team, then it may randomly hit "forbidden" error.

### LINKED ISSUE #6367: fly randomly hit "forbidden" error
## Summary

After we upgraded from 6.4.0 to 6.7.2, we keep receiving user escalation about "forbidden" error. The error usually happens with fly commands that require "member" role. The error occurs very randomly, and can be resolved by retry, sometime many retries needed.

## Steps to reproduce

I'm not fully sure how to reproduce the error, so I'm just describing how did I reproduce:

1. Having a cluster with 5 ATCs.
2. ATC auth with LDAP.
3. Create a team with a LDAP user as member; also configure a LDAP group as viewer. The user should be a member of the group. So that, the user has "member" via username and "viewer" role via group.
4. Repeat "set-pipeline" and "destroy-pipeline" to the team, then I hit the error.

## Expected results

Never hit "forbidden" error as the user takes "member" role.

## Actual results

Randomly hit "forbidden" error.

## Additional context

My best guess is that this is a bug introduced by #5897.

## Triaging info

* Concourse version: 6.7.2, but I guess 6.5.0+ should all have the problem
* Browser (if applicable): NA
* Did this used to work? 6.4.0 has no this problem

### PR DESCRIPTION
**Pull request by evanchaoli**
## What does this PR accomplish?

Bug Fix 

closes #6367 .

## Changes proposed by this PR:

This is a small logic error in function `hasPermission()`, and fixed the bad logic.

## Release Note

Fix an authorization bug, when a user has multiple roles in a team, then it may randomly hit "forbidden" error.

## Contributor Checklist

- [ ] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [ ] [Signed] all commits
- [ ] Added tests (Unit and/or Integration)
- [ ] Updated [Documentation]
- [ ] Added release note (Optional)

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

File: atc/api/accessor/accessor.go [hunk 1]
```diff
@@ -159,23 +159,31 @@ func (a *access) TeamNames() []string {
...
-			return false
+		has = has || a.hasPermissionCheckingRole(role)
+		if has {
+			return true
 		}
 	}
 	return false
 }
 
+func (a *access) hasPermissionCheckingRole(role string) bool {
```
Comment (muntac) [inline atc/api/accessor/accessor.go:172]:
```suggestion
func (a *access) hasRequiredRole(role string) bool {
```

File: atc/api/accessor/accessor.go [hunk 2]
```diff
@@ -159,23 +159,31 @@ func (a *access) TeamNames() []string {
 }
 
 func (a *access) hasPermission(roles []string) bool {
+	has := false
```
Comment (muntac) [inline atc/api/accessor/accessor.go:162]:
```suggestion
	allow := false
```

**Review by muntac — CHANGES_REQUESTED** [review]
@evanchaoli looks good! I suggested two name changes that may make things easier to read.

File: atc/api/accessor/accessor.go (hunk 2 above)
Comment (evanchaoli) [reply atc/api/accessor/accessor.go]: Renamed.

File: atc/api/accessor/accessor.go (hunk 1 above)
Comment (evanchaoli) [reply atc/api/accessor/accessor.go]: Renamed.

**Comment by evanchaoli** [comment]
@muntac Thanks for the suggestion of renaming. Please take a look again.

**Review by muntac — CHANGES_REQUESTED** [review]
@evanchaoli thanks! I was looking at the other tests and think we can make this test simpler. You could just pass the user and group role, then create the atc.TeamAuth struct in the test function itself.

```
func(requiredRole string, userRole string, groupRole string, expected bool) {
....
fakeTeam1.AuthReturns(atc.TeamAuth{
				userRole: map[string][]string{
					"users": {"some-connector:some-user-id"},
				},
				groupRole: map[string][]string{
				"groups": {"some-connector:some-group"},
				},
			})
...
Entry("user is member and group is viewer attempting owner action", "owner", "member", "viewer", false),
Entry("user is viewer and group is member attempting owner action", "owner", "viewer", "member", false),
....
```

**Comment by evanchaoli** [comment]
@muntac Revised the tests. Please take a look again.

BTW, #6336 is tiny fix of `fly`, and it could be a candidate for 673 if you ship, could you please review that one also?

**Review by muntac — APPROVED** [review]

**Pull request merged**